packages/@aws-cdk/aws-ec2/lib/vpc.ts

```diff
@@ -94,6 +94,21 @@ export interface IVpc extends IResource {
    */
   addVpnConnection(id: string, options: VpnConnectionOptions): VpnConnection;
 
+  /**
+   * Adds a new gateway endpoint to this VPC
+   */
+  addGatewayEndpoint(id: string, options: GatewayVpcEndpointOptions): GatewayVpcEndpoint
+
+  /**
+   * Adds a new S3 gateway endpoint to this VPC
+   */
+  addS3Endpoint(id: string, subnets?: SubnetSelection[]): GatewayVpcEndpoint
```

Comment:
Are we expecting this to eventually have over 100 methods, one for each AWS service? Feels like maybe we can just promote the `service` field from `GatewayVpcEndpointOptions` to a positional argument, and the it will read: `addGatewayEndpoint(GatewayVpcEndpointAwsService.S3)`.

Let's just leave `addGatewayEndpoint` and @deprecate the rest from `Vpc`